staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -218,6 +222,18 @@ func (o *ExposeServiceOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) e
 	return err
 }
 
+// Validate checks to the ExposeServiceOptions to see if there is sufficient information run the command.
+func (o *ExposeServiceOptions) Validate() error {
+	if len(o.Type) > 0 {
+		// Valid options for type are ExternalName, ClusterIP, NodePort, and LoadBalancer.
+		if o.Type != "ExternalName" && o.Type != "ClusterIP" && o.Type != "NodePort" && o.Type != "LoadBalancer" {
```

Comment:
Is there a list or constants of this somewhere instead of hard coding these? If not we might want to add a `ValidateServiceType` type of function.

Reply:
Looks like [these](https://github.com/kubernetes/kubernetes/blob/b497fa1d34ec15c4219d591e541c2cf811271357/staging/src/k8s.io/api/core/v1/types.go#L3916-L3934) are used in a [few places](https://cs.k8s.io/?q=ServiceTypeNodePort&i=nope&files=&repos=kubernetes/kubernetes).

Yea let's stick that in there and test it unless someone thinks of a better place.

Reply:
So i use the constants instead right? Earlier what i meant was that i didn't find all them as a list.